make lint fails installing golint.

**Is this a bug report or feature request?**

* Bug Report

**Deviation from expected behavior:**

in a fresh  git checkout of current master,  `make lint` fails at installing golint.
It does not even run the linting itself at all.

Details from MacOS:

```console
$ pwd
/Users/obnox/go/src/github.com/rook/rook
$ make lint
=== installing golint
go: github.com/golang/lint@v0.0.0-20210508222113-6edffad5e616 (matching github.com/golang/lint/golint@upgrade) requires github.com/golang/lint@v0.0.0-20210508222113-6edffad5e616: parsing go.mod:
	module declares its path as: golang.org/x/lint
	        but was required as: github.com/golang/lint
make[1]: *** [/Users/obnox/go/src/github.com/rook/rook/.cache/tools/linux_arm64/golint] Error 1
make: *** [lint] Error 2
$ 

```

Similar output on Linux.


**Expected behavior:**

`make lint` should install golint (in a local temporary location) and run the actual linting.



**How to reproduce it (minimal and precise):**

* `git clone https://github.com/rook/rook.git`
* `cd rook`
* `make lint`
* see error messages.

**File(s) to submit**:

none


**Logs to submit**:

none, as this is for a developer environment.
see paste of error output above for details.

**Cluster Status to submit**:

not applicable - developer environment.

**Output of kubectl commands, if necessary**:

not applicable - developer environment.


**Environment**:

*developer environment
*  OS: MacOS 15.1.1 and Fedora Linux 41 
* n/a
* n/a
* current master branch
* n/a
* n/a
* none
* n/a

after changing `github.com/golang/lint/golint`to `golang.org/x/lint/golint` in `build/makelib/golang.mk`, it seems to install correctly, but running `go.lint`still fails:

```console
$ make go.lint
=== installing golint
go: module github.com/golang.org/x/lint/golint: git ls-remote -q origin in /Users/obnox/go/src/github.com/rook/rook/.cache/tools/linux_arm64/tmp/pkg/mod/cache/vcs/d318ec1ee8ba683a4b8bf7eb3a1f4cbabaf3ae4dec5355cc0078c985fb0aa7c1: exit status 128:
	fatal: could not read Username for 'https://github.com': terminal prompts disabled
Confirm the import path was entered correctly.
If this is a private repository, see https://golang.org/doc/faq#git_https for additional information.
make: *** [/Users/obnox/go/src/github.com/rook/rook/.cache/tools/linux_arm64/golint] Error 1

$ 
```

--- a/Makefile
+++ b/Makefile
@@ -149,10 +149,6 @@ check test: ## Runs unit tests.
 test-integration: ## Runs integration tests.
 	@$(MAKE) go.test.integration
 
-lint: ## Check syntax and styling of go sources.
-	@$(MAKE) go.init
-	@$(MAKE) go.lint
-
 vet: ## Runs lint checks on go sources.
 	@$(MAKE) go.init
 	@$(MAKE) go.vet

--- a/build/makelib/golang.mk
+++ b/build/makelib/golang.mk
@@ -64,7 +64,6 @@ endif
 GOPATH := $(shell go env GOPATH)
 
 # setup tools used during the build
-GOLINT := $(TOOLS_HOST_DIR)/golint
 GOJUNIT := $(TOOLS_DIR)/go-junit-report
 
 GO := go
@@ -144,11 +143,6 @@ go.test.integration: $(GOJUNIT)
 	CGO_ENABLED=$(CGO_ENABLED_VALUE) $(GOHOST) test -v -timeout 7200s $(GO_TEST_FLAGS) $(GO_STATIC_FLAGS) $(GO_INTEGRATION_TEST_PACKAGES) $(TEST_FILTER_PARAM) 2>&1 | tee $(GO_TEST_OUTPUT)/integration-tests.log
 	@cat $(GO_TEST_OUTPUT)/integration-tests.log | $(GOJUNIT) -set-exit-code > $(GO_TEST_OUTPUT)/integration-tests.xml
 
-.PHONY: go.lint
-go.lint: $(GOLINT)
-	@echo === go lint
-	@$(GOLINT) -set_exit_status=true $(GO_PACKAGES) $(GO_INTEGRATION_TEST_PACKAGES)
-
 .PHONY: go.vet
 go.vet:
 	@echo === go vet
@@ -184,12 +178,6 @@ go.mod.clean:
 	@echo === cleaning modules cache
 	@sudo rm -fr $(WORK_DIR)/cross_pkg
 	@$(GOHOST) clean -modcache
-
-$(GOLINT):
-	@echo === installing golint
-	@mkdir -p $(TOOLS_HOST_DIR)/tmp
-	@GOPATH=$(TOOLS_HOST_DIR)/tmp GOBIN=$(TOOLS_HOST_DIR) $(GOHOST) get github.com/golang/lint/golint
-	@rm -fr $(TOOLS_HOST_DIR)/tmp
 
 $(GOFMT):
 	@echo === installing gofmt$(GOFMT_VERSION)

